### PR TITLE
Glukkon::PlaySound_4447D0

### PR DESCRIPTION
**Pull request by Varrok**
🌔    🦅 🌍

### CODE REVIEW

File: Source/AliveLib/Glukkon.cpp [hunk 1]
```diff
@@ -252,7 +252,7 @@ void Glukkon::M_Walk_1_442D30()
         case 10:
         case 14:
             PlaySound_4447D0(0, this);
-            field_212++;
+            currentWalkPitch_212++;
```
Comment (paulsapps) [inline Source/AliveLib/Glukkon.cpp]: rename to be field_212_currentWalkPitch all of the "field_addr" will be removed by a script at the end

File: Source/AliveLib/Glukkon.cpp [hunk 2]
```diff
@@ -2565,9 +2565,91 @@ void Glukkon::GetOnPlatforms_444060()
...
 }
 
-void CC Glukkon::PlaySound_4447D0(int /*sndIdx*/, Glukkon* /*pGlukkon*/)
+void CC Glukkon::PlaySound_4447D0(int sndIdx, Glukkon* pGlukkon)
 {
-    NOT_IMPLEMENTED();
+    int volumeLeft, volumeRight;
+    BaseAnimatedWithPhysicsGameObject* pGlukkonBase = ( BaseAnimatedWithPhysicsGameObject * ) pGlukkon;
+
+    SfxDefinition stepSfx_554840[3] =
```
Comment (paulsapps) [inline Source/AliveLib/Glukkon.cpp]: should live outside of the function like in OG otherwise its stack scoped

File: Source/AliveLib/Glukkon.cpp [hunk 3]
```diff
@@ -2565,9 +2565,91 @@ void Glukkon::GetOnPlatforms_444060()
         (TCollisionCallBack)&BaseAliveGameObject::OnTrapDoorIntersection_408BA0);
 }
 
-void CC Glukkon::PlaySound_4447D0(int /*sndIdx*/, Glukkon* /*pGlukkon*/)
+void CC Glukkon::PlaySound_4447D0(int sndIdx, Glukkon* pGlukkon)
 {
-    NOT_IMPLEMENTED();
+    int volumeLeft, volumeRight;
+    BaseAnimatedWithPhysicsGameObject* pGlukkonBase = ( BaseAnimatedWithPhysicsGameObject * ) pGlukkon;
```
Comment (paulsapps) [inline Source/AliveLib/Glukkon.cpp]: use C++ static_cast

File: Source/AliveLib/Glukkon.cpp [hunk 4]
```diff
@@ -2565,9 +2565,91 @@ void Glukkon::GetOnPlatforms_444060()
...
+    if (pGlukkonBase->field_CC_sprite_scale == FP_FromInteger(1))
+    {
+        volumeRight = defaultSndIdxVol;
+    }
+    else // Glukkon in background layer? TODO figure out if this does actually happen
+    {
+        volumeRight = defaultSndIdxVol / 2;
+    }
+
+    CameraPos direction;
```
Comment (paulsapps) [inline Source/AliveLib/Glukkon.cpp]: just use CameraPos direction= whatever directly

File: Source/AliveLib/Glukkon.cpp [hunk 5]
```diff
@@ -2565,9 +2565,91 @@ void Glukkon::GetOnPlatforms_444060()
...
+            volumeRight = volumeRight - FP_GetExponent(percentHowFar * FP_FromInteger(volumeRight - ( volumeRight / 3 )));
+        }
+            break;
+        default:
+            return;
+    }
+
+    if (pGlukkonBase->field_CC_sprite_scale.fpValue == 0x8000) //TODO figure out if this does actually happen
+    {
+        volumeLeft = ( signed int ) Math_FixedPoint_Divide_496B70(( signed __int16 ) volumeLeft << 17, 0x30000) / 0x10000;
```
Comment (paulsapps) [inline Source/AliveLib/Glukkon.cpp]: remove casts and use actual fixed point math here

**Comment by Varrok** [comment]
Also, added PlaySound_444AF0 📦

File: Source/AliveLib/Glukkon.cpp [hunk 6]
```diff
@@ -2650,9 +2731,44 @@ void Glukkon::vOn_TLV_Collision_4404A0(Path_TLV* pTlv)
...
+    { 0u, 0u, 0u, 0u, 0, 0 }, //[Nothing]
+    { 0u, 0u, 0u, 0u, 0, 0 } //[Nothing]
+};
+
+
+void CC Glukkon::PlaySound_444AF0(unsigned __int8 sndIdx, __int16 volume, __int16 pitch, Glukkon* pGlukkon) //todo rename PlaySound_Gamespeak_444AF0
 {
-    NOT_IMPLEMENTED();
+    auto pGlukkonBase = static_cast< BaseAnimatedWithPhysicsGameObject* >( pGlukkon );
+    int calcedVolume;
```
Comment (paulsapps) [inline Source/AliveLib/Glukkon.cpp]: use int calcedVolume = whatever as before :P

File: Source/AliveLib/Glukkon.cpp [hunk 7]
```diff
@@ -2650,9 +2731,44 @@ void Glukkon::vOn_TLV_Collision_4404A0(Path_TLV* pTlv)
...
+        calcedVolume = gameSpeak_554858[sndIdx].field_3_default_volume;
+    }
+    if (pGlukkon)
+    {
+        if (pGlukkonBase->field_CC_sprite_scale == FP_FromDouble(0.5))
+        {
+            calcedVolume = FP_GetExponent(FP_FromInteger(calcedVolume * 2) / FP_FromInteger(3));
+        }
+    }
+    SFX_SfxDefinition_Play_4CA420(&gameSpeak_554858[sndIdx], (__int16) calcedVolume, pitch, pitch);
```
Comment (paulsapps) [inline Source/AliveLib/Glukkon.cpp]: remove cast

File: Source/AliveLib/Glukkon.cpp [hunk 8]
```diff
@@ -2650,9 +2731,44 @@ void Glukkon::vOn_TLV_Collision_4404A0(Path_TLV* pTlv)
...
+    { 0u, 0u, 0u, 0u, 0, 0 }, //[Nothing]
+    { 0u, 0u, 0u, 0u, 0, 0 }, //[Nothing]
+    { 0u, 0u, 0u, 0u, 0, 0 } //[Nothing]
+};
+
+
+void CC Glukkon::PlaySound_444AF0(unsigned __int8 sndIdx, __int16 volume, __int16 pitch, Glukkon* pGlukkon) //todo rename PlaySound_Gamespeak_444AF0
 {
-    NOT_IMPLEMENTED();
+    auto pGlukkonBase = static_cast< BaseAnimatedWithPhysicsGameObject* >( pGlukkon );
```
Comment (paulsapps) [inline Source/AliveLib/Glukkon.cpp]: Cast is redundant because this is a base type

File: Source/AliveLib/Glukkon.cpp [hunk 9]
```diff
@@ -2565,9 +2565,90 @@ void Glukkon::GetOnPlatforms_444060()
...
-    NOT_IMPLEMENTED();
+    { 0u, 8u, 36u, 25u, 1524, 1905 },
+    { 0u, 3u, 59u, 60u, 0, 254 },
+    { 0u, 3u, 72u, 120u, 0, 254 }
+};
+
+void CC Glukkon::PlaySound_4447D0(int sndIdx, Glukkon* pGlukkon)
+{
+    int volumeLeft, volumeRight;
+    auto pGlukkonBase = static_cast<BaseAnimatedWithPhysicsGameObject*>(pGlukkon);
```
Comment (paulsapps) [inline Source/AliveLib/Glukkon.cpp]: ditto - redundant cast, no need to cast to base types, like polymorphism do its thing

File: Source/AliveLib/Glukkon.cpp [hunk 10]
```diff
@@ -2650,9 +2730,41 @@ void Glukkon::vOn_TLV_Collision_4404A0(Path_TLV* pTlv)
     }
 }
 
-void CC Glukkon::PlaySound_444AF0(unsigned __int8 /*sndIdx*/, __int16 /*volume*/, __int16 /*pitch*/, Glukkon* /*pGlukkon*/)
+SfxDefinition gameSpeak_554858[15] =
+{
+    { 0u, 8u, 66u, 127u, 0, 0 }, //HEY!
```
Comment (paulsapps) [inline Source/AliveLib/Glukkon.cpp]: you could add an enum for this and then change the call sites of PlaySound_GameSpeak_444AF0 to use it

File: Source/AliveLib/Glukkon.cpp [hunk 11]
```diff
@@ -677,7 +677,7 @@ void Glukkon::M_Speak1_11_4437D0()
                 pEventSystem_5BC11C->PushEvent_4218D0(static_cast<GameSpeakEvents>(field_1EA_speak + 36)); // TODO: Type safe conversion
             }
 
-            PlaySound_444AF0(field_1EA_speak, 0, 0, 0);
+            PlaySound_GameSpeak_444AF0(static_cast<GlukkonSpeak>(field_1EA_speak), 0, 0, 0);
```
Comment (paulsapps) [inline Source/AliveLib/Glukkon.cpp]: Change the type field_1EA_speak of to GlukkonSpeak

**Pull request merged**